Sources/GoodStructs/Either&Result.swift

```diff
@@ -198,28 +201,28 @@ public enum GRResult<V, E> {
         }
     }
 
-    public func unwrapSuccess() throws -> V {
+    public func unwrapSuccess() throws(UnwrapError) -> Success {
         switch self {
         case .success(let value):
             return value
 
         default:
-            throw EitherResultError.unwrap
+            throw .unwrap
         }
     }
 
-    public func unwrapFailure() throws -> E {
+    public func unwrapFailure() throws(UnwrapError) -> Failure {
         switch self {
         case .failure(let value):
             return value
 
         default:
-            throw EitherResultError.unwrap
+            throw .unwrap
         }
     }
 }
 
-enum EitherResultError: Error {
+public enum UnwrapError: Error {
 
     case unwrap
```

Comment:
Nebolo by lepsie pomenovanie "noValue"? Popripade nevracat error vobec ale dat to optional?